SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -461,29 +463,31 @@ void CMeshSolver::DeformMesh(CGeometry **geometry, CNumerics **numerics, CConfig
 
   if (multizone) nodes->Set_BGSSolution_k();
 
-  /*--- Compute the stiffness matrix. ---*/
-  Compute_StiffMatrix(geometry[MESH_0], numerics, config);
-
-  /*--- Clean residual, we do not want an incremental solution. ---*/
-  SU2_OMP_PARALLEL
-  {
-    LinSysRes.SetValZero();
-  }
-
-  /*--- LinSysSol contains the non-transformed displacements in the periodic halo cells.
-   Hence we still need a communication of the transformed coordinates, otherwise periodicity
-   is not maintained. ---*/
+  /*--- Capture a few MPI dependencies for AD. ---*/
   geometry[MESH_0]->InitiateComms(geometry[MESH_0], config, COORDINATES);
   geometry[MESH_0]->CompleteComms(geometry[MESH_0], config, COORDINATES);
 
-  /*--- In the same way, communicate the displacements in the solver to make sure the halo
-   nodes receive the correct value of the displacement. ---*/
   InitiateComms(geometry[MESH_0], config, SOLUTION);
   CompleteComms(geometry[MESH_0], config, SOLUTION);
 
   InitiateComms(geometry[MESH_0], config, MESH_DISPLACEMENTS);
   CompleteComms(geometry[MESH_0], config, MESH_DISPLACEMENTS);
 
+  /*--- Compute the stiffness matrix, no point recording because we clear the residual. ---*/
+
+  const bool ActiveTape = AD::TapeActive();
+  AD::StopRecording();
+
+  Compute_StiffMatrix(geometry[MESH_0], numerics, config);
+
+  if (ActiveTape) AD::StartRecording();
+
+  /*--- Clear residual (loses AD info), we do not want an incremental solution. ---*/
+  SU2_OMP_PARALLEL
+  {
+    LinSysRes.SetValZero();
+  }
```

Comment:
I am having a third second look at this bit.
Although the results did not change from this memory usage tweak, I think some dependencies might be lost here as this is equivalent to assuming the stiffness matrix does not depend on the reference (undeformed) mesh coordinates.
Which is also an approximation made in SU2_DOT.